### ISSUE TITLE
GeoServer is randomly not deployed on its docker container

### ISSUE THREAD

**Issue by francbartoli** [GeoNode/geoserver-docker#19]
GeoServer deployment within Tomcat 9 fails on its docker container:

```
geoserver4geonode | 2018-03-08T11:21:24.601926990Z  java.lang.IllegalStateException: ContainerBase.addChild: start: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Catalina].StandardHost[localhost].StandardContext[/geoserver]]
geoserver4geonode | 2018-03-08T11:21:24.601931568Z 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:744)
geoserver4geonode | 2018-03-08T11:21:24.601934973Z 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:716)
geoserver4geonode | 2018-03-08T11:21:24.601943577Z 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:703)
geoserver4geonode | 2018-03-08T11:21:24.601948119Z 	at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1131)
geoserver4geonode | 2018-03-08T11:21:24.601951369Z 	at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1865)
geoserver4geonode | 2018-03-08T11:21:24.601954402Z 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
geoserver4geonode | 2018-03-08T11:21:24.601957709Z 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
geoserver4geonode | 2018-03-08T11:21:24.601960734Z 	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
geoserver4geonode | 2018-03-08T11:21:24.601963992Z 	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
geoserver4geonode | 2018-03-08T11:21:24.601966929Z 	at org.apache.catalina.startup.HostConfig.deployDirectories(HostConfig.java:1043)
geoserver4geonode | 2018-03-08T11:21:24.601969636Z 	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:433)
geoserver4geonode | 2018-03-08T11:21:24.601972204Z 	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1575)
geoserver4geonode | 2018-03-08T11:21:24.601975072Z 	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:308)
geoserver4geonode | 2018-03-08T11:21:24.601978135Z 	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
geoserver4geonode | 2018-03-08T11:21:24.601981289Z 	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:424)
geoserver4geonode | 2018-03-08T11:21:24.601984313Z 	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:367)
geoserver4geonode | 2018-03-08T11:21:24.601987449Z 	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:964)
geoserver4geonode | 2018-03-08T11:21:24.601990510Z 	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:839)
geoserver4geonode | 2018-03-08T11:21:24.601993414Z 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
geoserver4geonode | 2018-03-08T11:21:24.601996339Z 	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1425)
geoserver4geonode | 2018-03-08T11:21:24.601999727Z 	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1415)
geoserver4geonode | 2018-03-08T11:21:24.602002736Z 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
geoserver4geonode | 2018-03-08T11:21:24.602005655Z 	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
geoserver4geonode | 2018-03-08T11:21:24.602008412Z 	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:134)
geoserver4geonode | 2018-03-08T11:21:24.602011253Z 	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:941)
geoserver4geonode | 2018-03-08T11:21:24.602014193Z 	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:258)
geoserver4geonode | 2018-03-08T11:21:24.602017117Z 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
geoserver4geonode | 2018-03-08T11:21:24.602019991Z 	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:422)
geoserver4geonode | 2018-03-08T11:21:24.602026438Z 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
geoserver4geonode | 2018-03-08T11:21:24.602029511Z 	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:770)
geoserver4geonode | 2018-03-08T11:21:24.602032375Z 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
geoserver4geonode | 2018-03-08T11:21:24.602034858Z 	at org.apache.catalina.startup.Catalina.start(Catalina.java:671)
geoserver4geonode | 2018-03-08T11:21:24.602037395Z 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
geoserver4geonode | 2018-03-08T11:21:24.602039927Z 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
geoserver4geonode | 2018-03-08T11:21:24.602042472Z 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
geoserver4geonode | 2018-03-08T11:21:24.602045228Z 	at java.lang.reflect.Method.invoke(Method.java:498)
geoserver4geonode | 2018-03-08T11:21:24.602048110Z 	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:353)
geoserver4geonode | 2018-03-08T11:21:24.602050797Z 	at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:493)
geoserver4geonode | 2018-03-08T11:21:24.602053733Z 
geoserver4geonode | 2018-03-08T11:21:24.603042134Z 08-Mar-2018 11:21:24.602 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/geoserver] has finished in [16,470] ms
geoserver4geonode | 2018-03-08T11:21:24.603057870Z 08-Mar-2018 11:21:24.602 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/usr/local/tomcat/webapps/ROOT]
geoserver4geonode | 2018-03-08T11:21:24.664298566Z 08-Mar-2018 11:21:24.664 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/ROOT] has finished in [61] ms
geoserver4geonode | 2018-03-08T11:21:24.664490373Z 08-Mar-2018 11:21:24.664 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/usr/local/tomcat/webapps/examples]
geoserver4geonode | 2018-03-08T11:21:24.920701892Z 08-Mar-2018 11:21:24.917 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/examples] has finished in [252] ms
geoserver4geonode | 2018-03-08T11:21:24.921025654Z 08-Mar-2018 11:21:24.920 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/usr/local/tomcat/webapps/docs]
geoserver4geonode | 2018-03-08T11:21:24.944041063Z 08-Mar-2018 11:21:24.943 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/docs] has finished in [23] ms
geoserver4geonode | 2018-03-08T11:21:24.944287182Z 08-Mar-2018 11:21:24.944 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/usr/local/tomcat/webapps/manager]
geoserver4geonode | 2018-03-08T11:21:24.968808921Z 08-Mar-2018 11:21:24.968 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/manager] has finished in [24] ms
geoserver4geonode | 2018-03-08T11:21:24.969711134Z 08-Mar-2018 11:21:24.969 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deploying web application directory [/usr/local/tomcat/webapps/host-manager]
geoserver4geonode | 2018-03-08T11:21:24.995043308Z 08-Mar-2018 11:21:24.994 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/host-manager] has finished in [26] ms
```

**Comment by afabiani** [comment]
Fixed

**Comment by Melegy90** [comment]
Unfortunately I'm facing this same issue on ubuntu 18 bionic using the "testing" geonode docker for OS compatibility. First everything ran fine, but when I uploaded a couple of layers to Geoserver and the server froze, I had to reboot, ever since "Geoserver" won't work, tomcat can't find it's directory. the logs :

12-Jul-2020 16:04:34.332 WARNING [main] org.geoserver.platform.GeoServerExtensions.checkContext Extension lookup 'GeoServerSecurityProvider', but ApplicationContext is unset.
12-Jul-2020 16:04:34.333 WARNING [main] org.geoserver.platform.GeoServerExtensions.checkContext Extension lookup 'ExtensionFilter', but ApplicationContext is unset.
12-Jul-2020 16:04:34.333 WARNING [main] org.geoserver.platform.GeoServerExtensions.checkContext Extension lookup 'ExtensionProvider', but ApplicationContext is unset.
12-Jul-2020 16:04:34.333 WARNING [main] org.geoserver.platform.GeoServerExtensions.checkContext Extension lookup 'ExtensionFilter', but ApplicationContext is unset.
12-Jul-2020 16:05:14.894 INFO [main] org.apache.catalina.startup.HostConfig.deployDirectory Deployment of web application directory [/usr/local/tomcat/webapps/geoserver] has finished in [65,302] ms
12-Jul-2020 16:05:14.904 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8080"]
12-Jul-2020 16:05:14.953 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["ajp-nio-8009"]
12-Jul-2020 16:05:14.995 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in [66,647] milliseconds
---------------------------------------------------------------------------------------------------------------

But when I actually navigate to the /usr/local/ directory, there is no tomcat folder to begin with.
